tools/src/main/scala/akka/contrib/persistence/mongodb/MigrateToSuffixedCollections.scala

```diff
@@ -31,176 +32,183 @@ class MigrateToSuffixedCollections(system: ActorSystem, config: Config) extends
-                      i + 1
-                    } recover {
-                      case _: Throwable =>
-                        logger.warn(s"Errors occurred when trying to insert record in '$collectionName' snapshot")
-                        i
-                    } getOrElse (i)
-                } match { case ssTotOk => (ok + ssTotOk, tot + cnt) }
-            } match {
-              case (inserted, count) =>
-                logger.info(s"$inserted/$count records were inserted into '$collectionName' snapshot")
-                tempDbObjects.foldLeft(0L) {
-                  case (ok, tdbo) =>
-                    val query = pidQuery(tdbo.get("_id").toString)
-                    Try { snaps.remove(query, snapsWriteConcern) } map { r =>
-                      ok + r.getN
-                    } recover {
-                      case _: Throwable =>
-                        logger.warn(s"Errors occurred when trying to remove records, previously copied to '$collectionName' snapshot, from '${settings.SnapsCollection}' snapshot")
-                        ok
-                    } getOrElse (ok)
-                } match {
-                  case removed =>
-                    logger.info(s"$removed/$count records, previously copied to '$collectionName' snapshot, were removed from '${settings.SnapsCollection}' snapshot")
-                    if (removed < inserted) (done + removed, ignored) else (done + inserted, ignored)
-                }
-            }
-          } else {
-            tempDbObjects.foldLeft(0L) {
-              case (tot, tdbo) =>
-                val query = pidQuery(tdbo.get("_id").toString)
-                val cnt = snaps.count(query).toLong
-                tot + cnt
-            } match { case ssTotIgnored => 
-                logger.info(s"$ssTotIgnored records were ignored and remain in '${settings.SnapsCollection}' journal")
-                (done, ignored + ssTotIgnored)
-            }
-          }
+  /**
+   * migrate records from an origin collection to some new collection and returns the total amount of migrated records
+   */
+  private[this] def migrateRecords(tempDbObjects: Seq[DBObject],
+                                   originCollection: MongoCollection,
+                                   newCollection: MongoCollection,
+                                   newCollectionName: String,
+                                   writeConcern: WriteConcern): Long = {
+    // first step: we insert records in new suffixed collection, foldLeft methods are only here for counting
+    tempDbObjects.foldLeft(0L, 0L) {
+      case ((ok, tot), tdbo) =>
+        val query = pidQuery(tdbo)
+        val cnt = originCollection.count(query).toLong
+        originCollection.find(query).foldLeft(0L) {
+          insertIntoCollection(newCollection, newCollectionName, writeConcern)
+        } match { case ssTotOk => (ok + ssTotOk, tot + cnt) }
+    } match {
```

Comment:
And again with this match, I think it would be simpler (less nested parens) to use tuple destructuring like:

```scala
val (inserted,count) = tempDbObjects.foldLeft((0,0)){ ...
}

logger.info(s"$inserted/$count records were inserted ...")

val removed = tempDbObjects.foldLeft(0L)(removeFromCollection(...))
logger.info(s"$removed/$count ...")
if (removed < inserted) removed else inserted
```